packages/sdk/src/fee/FeeEstimator.ts

```diff
@@ -0,0 +1,164 @@
+import { Psbt } from "bitcoinjs-lib";
+import { Buffer } from "buffer";
+import { MAXIMUM_FEE } from "../constants";
+import { OrditSDKError } from "../errors";
+import { getNetwork, getScriptType } from "../utils";
+import type { AddressFormat } from "../addresses/types";
+import type { Network } from "../config/types";
+import type { FeeEstimatorOptions } from "./types";
+
+class FeeEstimator {
+  protected fee = 0;
+  protected feeRate: number;
+  protected network: Network;
+  protected psbt: Psbt;
+  protected witness?: Buffer[] = [];
+  protected virtualSize = 0;
+  protected weight = 0;
+
+  constructor({ feeRate, network, psbt, witness }: FeeEstimatorOptions) {
+    if (feeRate < 0 || !Number.isSafeInteger(feeRate)) {
+      throw new OrditSDKError("Invalid feeRate");
+    }
+
+    this.feeRate = +feeRate; // convert decimal to whole number that might have passed Number.isSafeInteger check due to precision loss
+    this.network = network;
+    this.witness = witness || [];
+    this.psbt = psbt || new Psbt({ network: getNetwork(this.network) });
+  }
+
+  get data() {
+    return {
+      fee: this.fee,
+      virtualSize: this.virtualSize,
+      weight: this.weight,
+    };
+  }
+
+  private sanityCheckFee() {
+    if (this.fee > MAXIMUM_FEE) {
+      throw new OrditSDKError("Error while calculating fees");
+    }
+  }
```

Comment:
@iamcrazycoder Why does this require its own method if it's only used within `calculateNetworkFee`? Can add a code comment indicating what it is for without a new method

Reply:
The idea was to eventually convert it to a static fn that can be reused by non-derived classes to check the calculated fee.

Reply:
As non-derived classes are not calling this currently, I will move this into `calculateNetworkFee` to reduce function complexity.